src/ImageSearchModal.jsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { Button, ListGroup, ListGroupItem, Modal } from 'patternfly-react';
+
+import { ErrorNotification } from './Notification.jsx';
+import * as utils from './util.js';
+import varlink from './varlink.js';
+import cockpit from 'cockpit';
+
+import '../lib/form-layout.less';
+import './ImageSearchModal.css';
+
+const _ = cockpit.gettext;
+
+export class ImageSearchModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            imageIdentifier: '',
+            imageList: [],
+            searchInProgress: false,
+        };
+        this.onDownloadClicked = this.onDownloadClicked.bind(this);
+        this.onItemSelected = this.onItemSelected.bind(this);
+        this.onSearchTriggered = this.onSearchTriggered.bind(this);
+        this.onValueChanged = this.onValueChanged.bind(this);
+    }
+
+    componentDidMount() {
+        this._isMounted = true;
+    }
+
+    componentWillUnmount() {
+        this._isMounted = false;
+    }
+
+    onDownloadClicked() {
+        let selectedImageName = this.state.imageList[this.state.selected].name;
+
+        this.props.close();
+        this.props.downloadImage(selectedImageName);
+    }
+
+    onItemSelected(key) {
+        this.setState({ selected: key });
+    }
+
+    onSearchTriggered() {
+        if (!this._isMounted)
+            return;
+
+        varlink.call(utils.PODMAN_ADDRESS, "io.podman.SearchImage", { name: this.state.imageIdentifier })
+                .then(reply => this.setState({ imageList: reply.images || [], searchInProgress: false }))
+                .catch(ex => {
+                    this.setState({
+                        searchInProgress: false,
+                        dialogError: _("Failed to search for new images"),
+                        dialogErrorDetail: cockpit.format("$0: $1", ex.error, ex.parameters && ex.parameters.reason)
+                    });
+                });
```

Comment:
Unfortunately, this is where our simple handling of varlink calls fails. Cancelling the dialog doesn't cancel this call, which leads to
```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in ImageSearchModal (created by Images)
```

We have this problem in other components as well, but they are on the main page and don't get unmounted during normal operation.

We could
* Check for `this._isMounted` in the callbacks, which is annyoing
* Keep track of a varlink connection and `.close()` it on unmount. This would trigger all `.catch` handlers though. Not sure if we want that?
* Use something like redux to get all of the state handling out of components. Phew.

Reply:
I wouldn't go with redux just because of this PR, it's bug change.
I would do the this._isMounted trick. I already do it anyway.
But good catch thanks.